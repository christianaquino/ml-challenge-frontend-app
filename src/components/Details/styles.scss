@use 'sass:map';
@use '../../global.scss' as *;

.details {
  background-color: white;
  grid-area: main;
  padding: 0 32px;
  display: grid;
  grid-template-areas:
    'image condition condition condition condition'
    'image title title title title'
    'image price price price price'
    'image buybutton buybutton ... ...'
    'image ... ... ... ...'
    'titledesc titledesc titledesc titledesc titledesc'
    'description description description ... ...';

  &--imagecontainer {
    grid-area: image;
    max-height: 680px;
  }

  &--image {
    max-height: 680px;
  }

  &--condition {
    grid-area: condition;
    padding-top: 32px;
    padding-bottom: 16px;
    color: map.get($colors, 'ml-gray-3');
  }

  &--price {
    grid-area: price;
    font-size: 46px;
    padding: 32px 0;
  }

  &--title {
    font-size: 24px;
    grid-area: title;
  }

  &--buybutton {
    grid-area: buybutton;
    padding-right: 32px;
  }

  &--titledesc {
    font-size: 28px;
    grid-area: titledesc;
    padding-bottom: 32px;
  }

  &--description {
    font-size: 16px;
    grid-area: description;
    color: map.get($colors, 'ml-gray-3');
    padding-bottom: 32px;
  }
}
