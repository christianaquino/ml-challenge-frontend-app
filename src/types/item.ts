export interface IPrice {
  currency: string,
  amount: Number
}

export interface IItem {
  id: string,
  picture: string,
  title: string,
  price: IPrice
  condition: string
  free_shipping: boolean
}

export interface IItemDetails extends IItem {
  sold_quantity: Number,
  description: string
}
