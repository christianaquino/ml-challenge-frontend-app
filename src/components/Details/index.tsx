import { IItemDetails } from '../../types/item';

import '../../global.scss';
import './styles.scss';

interface IDetailsProps {
    itemDetails: IItemDetails
}

function Details({ itemDetails }: IDetailsProps) {
  return (
    <div className="details">
      <div className="details--imagecontainer">
        <img className="details--image" src={itemDetails.picture} alt={itemDetails.title} />
      </div>
      <div className="details--condition">{`${itemDetails.condition} - ${itemDetails.sold_quantity} vendidos`}</div>
      <div className="details--title">{itemDetails.title}</div>
      <div className="details--price">{`${itemDetails.price.currency} ${itemDetails.price.amount}`}</div>
      <button type="button" className="button-primary details--buybutton">Comprar</button>
      <div className="details--titledesc">Descripci√≥n del producto</div>
      <div className="details--description">{itemDetails.description}</div>
    </div>
  );
}

export default Details;
