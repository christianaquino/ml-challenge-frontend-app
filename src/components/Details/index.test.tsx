import { render } from '@testing-library/react';
import Details from '.';
import { IItemDetails } from '../../types/item';

describe('Details component', () => {
  test('Renders correctly', () => {
    const itemDetails: IItemDetails = {
      id: 'MLC8752432',
      description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys'
                          + 'standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make'
                          + 'a type specimen book',
      title: 'IPhone 12 Pro',
      sold_quantity: 20,
      free_shipping: true,
      price: {
        currency: 'CLP',
        amount: 1560000,
      },
      condition: 'Nuevo',
      picture: '//iphone.jpg',
    };

    const { asFragment } = render(<Details itemDetails={itemDetails} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
