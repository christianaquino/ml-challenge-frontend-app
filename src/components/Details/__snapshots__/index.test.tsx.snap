// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="details"
  >
    <div
      class="details--imagecontainer"
    >
      <img
        alt="IPhone 12 Pro"
        class="details--image"
        src="//iphone.jpg"
      />
    </div>
    <div
      class="details--condition"
    >
      Nuevo - 20 vendidos
    </div>
    <div
      class="details--title"
    >
      IPhone 12 Pro
    </div>
    <div
      class="details--price"
    >
      CLP 1560000
    </div>
    <button
      class="button-primary details--buybutton"
      type="button"
    >
      Comprar
    </button>
    <div
      class="details--titledesc"
    >
      Descripci√≥n del producto
    </div>
    <div
      class="details--description"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrysstandard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to makea type specimen book
    </div>
  </div>
</DocumentFragment>
`;
